packages/react/src/createReactQueryHooks.tsx

```diff
@@ -60,6 +61,25 @@ interface UseTRPCMutationOptions<TInput, TError, TOutput>
   extends UseMutationOptions<TOutput, TError, TInput>,
     TRPCUseQueryBaseOptions {}
 
+type Override<A, B> = {
+  [K in keyof A]: K extends keyof B ? B[K] : A[K];
+};
+
+type UseTRPCMutationResult<
+  TInput,
+  TOutput,
+  TError,
+  TOptions = UseTRPCMutationOptions<TOutput, TError, never>,
+> = TInput extends null | undefined
+  ? Override<
+      UseMutationResult<TOutput, TError, TInput>,
+      {
+        mutate: (variables?: null, options?: TOptions) => TOutput;
+        mutateAsync: (variables?: null, options?: TOptions) => Promise<TOutput>;
```

Comment:
Probably makes sense to do:

```suggestion
        mutate: (variables?: TInput, options?: TOptions) => TOutput;
        mutateAsync: (variables?: TInput, options?: TOptions) => Promise<TOutput>;
```

Reply:
That'd fix nullish (if it was a problem which I'm not sure of)